examples/datasource-streaming-backend-websocket/streaming-backend-websocket-plugin/src/datasource.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  CoreApp,
+  DataQueryRequest,
+  DataQueryResponse,
+  DataSourceInstanceSettings,
+  LiveChannelScope,
+} from '@grafana/data';
+import { DataSourceWithBackend, getGrafanaLiveSrv } from '@grafana/runtime';
+
+import { defaults } from 'lodash';
+import { Observable, merge } from 'rxjs';
+
+import { MyQuery, MyDataSourceOptions, DEFAULT_QUERY } from './types';
+
+export class DataSource extends DataSourceWithBackend<MyQuery, MyDataSourceOptions> {
+  constructor(instanceSettings: DataSourceInstanceSettings<MyDataSourceOptions>) {
+    super(instanceSettings);
+  }
+
+  query(request: DataQueryRequest<MyQuery>): Observable<DataQueryResponse> {
+    const observables = request.targets.map((target, index) => {
+      const query = defaults(target, DEFAULT_QUERY);
+
+      return getGrafanaLiveSrv().getDataStream({
+        addr: {
+          scope: LiveChannelScope.DataSource,
+          namespace: this.uid,
+          path: `my-ws/custom-${query.lowerLimit}-${query.upperLimit}`, // this will allow each new query to create a new connection
```

Comment:
The path doesn't need to include the limit, but the path needs to change every time the query changes. A new query should create a new connection and the way of doing it is by creating a new path. A prettier way of doing the same thing is by creating a hash from the query parameters (that is what loki does, e.g.: 
https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/streaming.ts#L68

https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/streaming.ts#L21-L28)

I did like this because it seems simpler